bb.edn

```diff
@@ -51,16 +51,52 @@
                      :task (shell {:dir "test-hello-world"}
                                   "bb native-image-test")}
 
-  bump-version (do (load-file "scripts/bump_version.clj")
-                   (shell "git add resources/clj-easy/graal-build-time-version.txt")
-                   (shell "git commit -m 'Bump version'")
-                   (bs/refresh-version))
+  change-log-check {:doc "Checks that change log is ready for publish"
+                    :task (let [log (slurp "CHANGELOG.md")
+                                new-changes (last (re-find (re-pattern "(?ms)^## Unreleased *$(.*?)^##|\\z") log))]
+                            (if (not (and new-changes (re-find (re-pattern "(?ims)[a-z]") new-changes)))
+                              (do
+                                (println "FAIL: Change log must contain Unreleased section with some text.")
+                                (System/exit 1))
+                              (println "PASS: Change log Unreleased section found with some text.")))}
 
-  tag (shell (str "git tag v" bs/version))
+  -change-log-update {:doc "Updates change log for current version"
+                      :depends [change-log-check]
+                      :task (do
+                              (println "Updating change log")
+                              (bs/refresh-version)
+                              (let [log (slurp "CHANGELOG.md")
```

Comment:
@lread Perhaps we can move some of the longer code fragments into build_shared.clj, what do you think?

Reply:
Thanks for the review. Sure, I think that makes sense.  I shall make it so.

Side bar: I noticed that could not use regex reader conditional `#"myregexhere"` bb in bb `:task` code. Hence the usage of `re-pattern`.

Reply:
Yep, that's expected, EDN doesn't support it.

Reply:
Right, of course!